src/realm/query_expression.hpp

```diff
@@ -1172,11 +1177,11 @@ class Value : public ValueBase, public Subexpr2<T> {
 public:
     Value()
     {
-        init(false, ValueBase::default_size, T());
+        init(false, 1, T());
```

Comment:
This will break the performance of `Columns<int64>::evaluate_internal`. It will read a chunk of data, but only use the first element.

Reply:
Thanks for catching this and fixing it. I added a Int vs double column query benchmark which hits this code path to show that that the performance is not negatively affected in this case:
master:
```
QueryIntsVsDoubleColumns (MemOnly, EncryptionOff):   min  25.03ms     max  28.08ms     median  25.60ms     avg  25.76ms     stddev    887us
QueryIntsVsDoubleColumns (MemOnly, EncryptionOn):    min  25.48ms     max  27.53ms     median  26.02ms     avg  26.03ms     stddev    592us
QueryIntsVsDoubleColumns (Full   , EncryptionOff):   min  25.36ms     max  26.67ms     median  25.94ms     avg  25.82ms     stddev    451us
QueryIntsVsDoubleColumns (Full   , EncryptionOn):    min  25.37ms     max  26.89ms     median  25.75ms     avg  25.83ms     stddev    468us
```
this branch: 
```
QueryIntsVsDoubleColumns (MemOnly, EncryptionOff):   min  24.43ms     max  25.73ms     median  24.92ms     avg  24.90ms     stddev    340us
QueryIntsVsDoubleColumns (MemOnly, EncryptionOn):    min  24.71ms     max  26.24ms     median  24.89ms     avg  25.02ms     stddev    453us
QueryIntsVsDoubleColumns (Full   , EncryptionOff):   min  24.53ms     max  27.03ms     median  24.84ms     avg  25.17ms     stddev    947us
QueryIntsVsDoubleColumns (Full   , EncryptionOn):    min  24.65ms     max  25.19ms     median  24.85ms     avg  24.85ms     stddev    195us
```